RoaringBitmap/src/main/java/org/roaringbitmap/BitmapContainer.java

```diff
@@ -414,6 +414,16 @@ public void deserialize(DataInput in) throws IOException {
     }
   }
 
+  @Override
+  public void deserialize(ByteBuffer byteBuffer) throws IOException {
+    this.cardinality = 0;
```

Comment:
This is inefficient compared to the way this is done elsewhere (converting to `LongBuffer` and calling `LongBuffer.get(long[])`) because the bounds checks get eliminated that way. It also shouldn't be necessary to calculate the cardinality, should just have it stored in the serialised format.